dot/state/block.go

```diff
@@ -366,11 +363,7 @@ func (bs *BlockState) GetBlockBody(hash common.Hash) (*types.Body, error) {
 
 // SetBlockBody will add a block body to the db
 func (bs *BlockState) SetBlockBody(hash common.Hash, body *types.Body) error {
-	bs.lock.Lock()
-	defer bs.lock.Unlock()
```

Comment:
I'm a confused why some locks were removed (lines 333, and 369), but others added in different spots (add at line 423, 495 and 593).  Doesn't seem to have any race conditions now, so this is working.  How do we determine where locks are needed?

Reply:
the db operations are already threadsafe (chaindb has locks for all db.Put, db.Get, etc.), and same for the blocktree, so I only added locks for functions that did multiple operations in them where some  variables need to not change until the function is done, for example AddBlock requires no other blocks to be added until it's finished